# 🚀 머신러닝 실습 : 고객 구매 데이터로 성별 예측 모델링 (분류 문제)

* 주어진 데이터는 백화점 고객의 1년 간 구매 데이터입니다.
* 고객 3,500명에 대한 학습용 데이터(y.csv, X.csv)를 이용하여 성별예측 모형을 만들어보세요.
* 모델의 성능은 자유롭게 측정해봅니다!

## [실습 프로세스]
1. 데이터 불러오기  
2. 데이터 탐색
3. 데이터 전처리  
4. 학습/테스트 데이터 분리  
5. 모델 선택 및 학습  
6. 예측 및 평가  


<br/>

---

<br/>
<br/>

# 0. 라이브러리 불러오기

* 라이브러리를 가져와서 과정을 준비합니다

In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

import warnings
warnings.filterwarnings(action='ignore')

<br/>

---

<br/>
<br/>

# 1. 데이터 불러오기
* 데이터를 가져와서 과정을 준비합시다.
- 인코딩 방식은 'euc-kr' 을 활용하세요.
- 데이터 출처 : 한국데이터산업진흥원 빅데이터분석기사 실기 공개 예시 문항

- 독립 변수 데이터셋 : ./data/X.csv
- 종속 변수 데이터셋 : ./data/y.csv


데이터 파일을 불러옵니다. 보통 CSV 파일을 pandas로 읽어옵니다.

In [4]:
# import os
# # 노트북 파일이 있는 폴더로 이동 (예시)
# os.chdir(r'C:\githome\hipython_rep')

# # 변경 후 확인
# print("변경 후:", os.getcwd())

In [5]:
X = pd.read_csv('./data/X.csv', encoding='euc-kr')
y = pd.read_csv('./data/y.csv', encoding='euc-kr')

In [6]:
X.head()

,cust_id,총구매액,최대구매액,환불금액,주구매상품,주구매지점,내점일수,내점당구매건수,주말방문비율,구매주기
0,0,68282840,11264000,6860000.0,기타,강남점,19,3.894737,0.527027,17
1,1,2136000,2136000,300000.0,스포츠,잠실점,2,1.500000,0.000000,1
2,2,3197000,1639000,NaN,남성 캐주얼,관악점,2,2.000000,0.000000,1
3,3,16077620,4935000,NaN,기타,광주점,18,2.444444,0.318182,16
4,4,29050000,24000000,NaN,보석,본 점,2,1.500000,0.000000,85


In [7]:
y.head()

,cust_id,gender
0,0,0
1,1,0
2,2,1
3,3,1
4,4,0


<br/>

---

<br/>
<br/>

# 2. 데이터 탐색하기
* 데이터를 이해할 수 있도록 탐색과정을 수행해봅시다.


데이터의 상위 몇 개 행을 출력하여 전체 구조를 미리 확인합니다.

In [8]:
X.head()

,cust_id,총구매액,최대구매액,환불금액,주구매상품,주구매지점,내점일수,내점당구매건수,주말방문비율,구매주기
0,0,68282840,11264000,6860000.0,기타,강남점,19,3.894737,0.527027,17
1,1,2136000,2136000,300000.0,스포츠,잠실점,2,1.500000,0.000000,1
2,2,3197000,1639000,NaN,남성 캐주얼,관악점,2,2.000000,0.000000,1
3,3,16077620,4935000,NaN,기타,광주점,18,2.444444,0.318182,16
4,4,29050000,24000000,NaN,보석,본 점,2,1.500000,0.000000,85


In [9]:
y.head()

,cust_id,gender
0,0,0
1,1,0
2,2,1
3,3,1
4,4,0



데이터의 요약 정보나 통계 정보를 출력해 변수들의 유형과 분포를 확인합니다.

In [10]:
X.info()

<class 'pandas.DataFrame'>
RangeIndex: 3500 entries, 0 to 3499
Data columns (total 10 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   cust_id  3500 non-null   int64  
 1   총구매액     3500 non-null   int64  
 2   최대구매액    3500 non-null   int64  
 3   환불금액     1205 non-null   float64
 4   주구매상품    3500 non-null   str    
 5   주구매지점    3500 non-null   str    
 6   내점일수     3500 non-null   int64  
 7   내점당구매건수  3500 non-null   float64
 8   주말방문비율   3500 non-null   float64
 9   구매주기     3500 non-null   int64  
dtypes: float64(3), int64(5), str(2)
memory usage: 273.6 KB


In [11]:
X.describe()

,cust_id,총구매액,최대구매액,환불금액,내점일수,내점당구매건수,주말방문비율,구매주기
count,3500.000000,3.500000e+03,3.500000e+03,1.205000e+03,3500.000000,3500.000000,3500.000000,3500.000000
mean,1749.500000,9.191925e+07,1.966424e+07,2.407822e+07,19.253714,2.834963,0.307246,20.958286
std,1010.507298,1.635065e+08,3.199235e+07,4.746453e+07,27.174942,1.912368,0.289752,24.748682
min,0.000000,-5.242152e+07,-2.992000e+06,5.600000e+03,1.000000,1.000000,0.000000,0.000000
25%,874.750000,4.747050e+06,2.875000e+06,2.259000e+06,2.000000,1.666667,0.027291,4.000000
50%,1749.500000,2.822270e+07,9.837000e+06,7.392000e+06,8.000000,2.333333,0.256410,13.000000
75%,2624.250000,1.065079e+08,2.296250e+07,2.412000e+07,25.000000,3.375000,0.448980,28.000000
max,3499.000000,2.323180e+09,7.066290e+08,5.637530e+08,285.000000,22.083333,1.000000,166.000000


In [12]:
X.isnull().sum()

cust_id       0
총구매액          0
최대구매액         0
환불금액       2295
주구매상품         0
주구매지점         0
내점일수          0
내점당구매건수       0
주말방문비율        0
구매주기          0
dtype: int64

In [13]:
y.info()

<class 'pandas.DataFrame'>
RangeIndex: 3500 entries, 0 to 3499
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   cust_id  3500 non-null   int64
 1   gender   3500 non-null   int64
dtypes: int64(2)
memory usage: 54.8 KB


In [14]:
y.describe()

,cust_id,gender
count,3500.000000,3500.000000
mean,1749.500000,0.376000
std,1010.507298,0.484449
min,0.000000,0.000000
25%,874.750000,0.000000
50%,1749.500000,0.000000
75%,2624.250000,1.000000
max,3499.000000,1.000000


In [15]:
y.isnull().sum()

cust_id    0
gender     0
dtype: int64

<br/>

---

<br/>
<br/>

# 3. 데이터 전처리
* 전처리 과정을 통해서 머신러닝에 사용할 수 있는 형태의 데이터 준비


필요한 라이브러리를 불러옵니다.
- 인코딩 : LabelEncoder
- 데이터 표준화 : StandardScaler

In [16]:
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler

* 단순히 1부터의 숫자를 부여한 'cust_id'를 수치형 변수로 받아들이면, 결과가 왜곡될 수 있으니 컬럼을 제거합니다.

In [17]:
X = X.drop('cust_id', axis=1)

In [18]:
X.head()

,총구매액,최대구매액,환불금액,주구매상품,주구매지점,내점일수,내점당구매건수,주말방문비율,구매주기
0,68282840,11264000,6860000.0,기타,강남점,19,3.894737,0.527027,17
1,2136000,2136000,300000.0,스포츠,잠실점,2,1.500000,0.000000,1
2,3197000,1639000,NaN,남성 캐주얼,관악점,2,2.000000,0.000000,1
3,16077620,4935000,NaN,기타,광주점,18,2.444444,0.318182,16
4,29050000,24000000,NaN,보석,본 점,2,1.500000,0.000000,85


In [19]:
y = y.drop('cust_id', axis=1)

In [20]:
y.head()

,gender
0,0
1,0
2,1
3,1
4,0


- 데이터에 결측치가 있는지 확인해보세요


In [21]:
X.isnull().sum()

총구매액          0
최대구매액         0
환불금액       2295
주구매상품         0
주구매지점         0
내점일수          0
내점당구매건수       0
주말방문비율        0
구매주기          0
dtype: int64

In [22]:
y.isnull().sum()

gender    0
dtype: int64

- 결측치에 0으로 채워 넣어 모델 학습에 지장이 없도록 합니다.

In [23]:
X['환불금액'] = X['환불금액'].fillna(0)

In [24]:
X.isnull().sum()

총구매액       0
최대구매액      0
환불금액       0
주구매상품      0
주구매지점      0
내점일수       0
내점당구매건수    0
주말방문비율     0
구매주기       0
dtype: int64


문자형 범주 데이터를 숫자로 바꾸기 위한 인코딩을 수행합니다.

In [25]:
X = pd.get_dummies(
    X,
    columns=['주구매상품', '주구매지점'],
    drop_first=True
)

In [26]:
X

,총구매액,최대구매액,환불금액,내점일수,내점당구매건수,주말방문비율,구매주기,주구매상품_가구,주구매상품_건강식품,주구매상품_골프,...,주구매지점_안양점,주구매지점_영등포점,주구매지점_울산점,주구매지점_인천점,주구매지점_일산점,주구매지점_잠실점,주구매지점_전주점,주구매지점_창원점,주구매지점_청량리점,주구매지점_포항점
0,68282840,11264000,6860000.0,19,3.894737,0.527027,17,False,False,False,...,False,False,False,False,False,False,False,False,False,False
1,2136000,2136000,300000.0,2,1.500000,0.000000,1,False,False,False,...,False,False,False,False,False,True,False,False,False,False
2,3197000,1639000,0.0,2,2.000000,0.000000,1,False,False,False,...,False,False,False,False,False,False,False,False,False,False
3,16077620,4935000,0.0,18,2.444444,0.318182,16,False,False,False,...,False,False,False,False,False,False,False,False,False,False
4,29050000,24000000,0.0,2,1.500000,0.000000,85,False,False,False,...,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3495,3175200,3042900,0.0,1,2.000000,1.000000,0,False,False,True,...,False,False,False,False,False,False,False,False,False,False
3496,29628600,7200000,6049600.0,8,1.625000,0.461538,40,False,False,False,...,False,False,False,False,False,False,False,False,False,False
3497,75000,75000,0.0,1,1.000000,0.000000,0,False,False,False,...,False,False,False,False,False,False,False,True,False,False
3498,1875000,1000000,0.0,2,1.000000,0.000000,39,False,False,False,...,False,False,False,False,False,False,False,False,False,False


각 데이터에 표준화를 적용하여 데이터의 스케일(크기 차이)을 맞춰줍니다.
- 평균을 0, 표준편차를 1로 맞춰서 → 데이터가 정규 분포 형태로 변환되도록 하세요

In [27]:
scaler = StandardScaler()
X_scaled = pd.DataFrame(
    scaler.fit_transform(X),
    columns=X.columns
)

In [28]:
X.head()

,총구매액,최대구매액,환불금액,내점일수,내점당구매건수,주말방문비율,구매주기,주구매상품_가구,주구매상품_건강식품,주구매상품_골프,...,주구매지점_안양점,주구매지점_영등포점,주구매지점_울산점,주구매지점_인천점,주구매지점_일산점,주구매지점_잠실점,주구매지점_전주점,주구매지점_창원점,주구매지점_청량리점,주구매지점_포항점
0,68282840,11264000,6860000.0,19,3.894737,0.527027,17,False,False,False,...,False,False,False,False,False,False,False,False,False,False
1,2136000,2136000,300000.0,2,1.500000,0.000000,1,False,False,False,...,False,False,False,False,False,True,False,False,False,False
2,3197000,1639000,0.0,2,2.000000,0.000000,1,False,False,False,...,False,False,False,False,False,False,False,False,False,False
3,16077620,4935000,0.0,18,2.444444,0.318182,16,False,False,False,...,False,False,False,False,False,False,False,False,False,False
4,29050000,24000000,0.0,2,1.500000,0.000000,85,False,False,False,...,False,False,False,False,False,False,False,False,False,False


In [29]:
scaler = StandardScaler()
y_scaled = pd.DataFrame(
    scaler.fit_transform(y),
    columns=y.columns
)

In [30]:
y.head()

,gender
0,0
1,0
2,1
3,1
4,0


<br/>

---

<br/>
<br/>

# 5-1. 모델링 - LogisticRegression

* 본격적으로 모델을 선언하고 학습시킵니다.


필요한 라이브러리를 불러옵니다.

In [31]:
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

모델을 선언하여 객체화시킵니다.

In [32]:
lr = LogisticRegression()


모델을 학습 데이터에 맞춰 학습시킵니다.

In [33]:

X_train,X_test,y_train,y_test = train_test_split(X, y, test_size=0.2)
X_train.shape ,X_test.shape ,y_train.shape ,y_test.shape

((2800, 71), (700, 71), (2800, 1), (700, 1))

In [47]:
lr_clf = LogisticRegression(
    penalty='l2',
    C=1.0,
    solver='liblinear',
    max_iter=100,
    random_state=0
)
lr_clf.fit(X_train, y_train) 
lr_pred = lr_clf.predict(X_test)
accuracy_score(y_test, lr_pred)

0.6514285714285715

<br/>

---

<br/>
<br/>

# 6-1. 예측 성능 확인해보기 - LogisticRegression

- 학습된 모델로 테스트 데이터에 대한 예측을 수행합니다.

- 학습시킨 모델의 성능을 알아봅니다
- 각 평가지표로 모델의 성능을 수치화하여 확인합니다.
- 필요한 라이브러리를 import 하고 성능을 확인해보세요 (정확도, 정밀도, 재현율, f1, confusion_matrix)

In [36]:
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import confusion_matrix
from sklearn.metrics import precision_score, recall_score

In [44]:
from sklearn.metrics import (
    confusion_matrix,
    accuracy_score,
    precision_score,
    recall_score,
    f1_score
)

def get_clf_eval(y_test, pred):
    cm = confusion_matrix(y_test, pred)
    acc = accuracy_score(y_test, pred)
    prec = precision_score(y_test, pred, zero_division=0)
    rec = recall_score(y_test, pred, zero_division=0)
    f1 = f1_score(y_test, pred, zero_division=0)

    print("Confusion Matrix")
    print(cm)
    print("-" * 30)
    print(f"Accuracy : {acc:.4f}")
    print(f"Precision: {prec:.4f}")
    print(f"Recall   : {rec:.4f}")
    print(f"F1-score : {f1:.4f}")

In [57]:
print("📌 Decision Tree 결과")
get_clf_eval(y_test, pred)

📌 Decision Tree 결과
Confusion Matrix
[[413  43]
 [177  67]]
------------------------------
Accuracy : 0.6857
Precision: 0.6091
Recall   : 0.2746
F1-score : 0.3785



<br/>

---

<br/>
<br/>

# 5-2. 모델링 - DecisionTreeClassifier

* 본격적으로 모델을 선언하고 학습시킵니다.


필요한 라이브러리를 불러옵니다.

모델을 선언하여 객체화시킵니다.

모델을 학습 데이터에 맞춰 학습시킵니다.

In [49]:
dt_clf = DecisionTreeClassifier(
    max_depth=4,          # 너무 깊어지는 과적합 방지
    min_samples_split=10, # 분할 최소 샘플
    min_samples_leaf=5,   # 리프 최소 샘플
    random_state=0
)
dt_clf.fit(X_train, y_train) 
dt_pred = dt_clf.predict(X_test)
accuracy_score(y_test, dt_pred)

0.6571428571428571



<br/>
<br/>

# 6-2. 예측 성능 확인해보기 - DecisionTreeClassifier

- 학습된 모델로 테스트 데이터에 대한 예측을 수행합니다.

- 학습시킨 모델의 성능을 알아봅니다
- 각 평가지표로 모델의 성능을 수치화하여 확인합니다.
- 필요한 라이브러리를 import 하고 성능을 확인해보세요 (정확도, 정밀도, 재현율, f1, confusion_matrix)

In [ ]:
print("📌 Decision Tree 결과")
get_clf_eval(y_test, dt_pred)

📌 Decision Tree 결과
Confusion Matrix
[[358  98]
 [142 102]]
------------------------------
Accuracy : 0.6571
Precision: 0.5100
Recall   : 0.4180
F1-score : 0.4595



<br/>

---

<br/>
<br/>

# 5-3. 모델링 - RandomForestClassifier

* 본격적으로 모델을 선언하고 학습시킵니다.



필요한 라이브러리를 불러옵니다.

In [50]:
from sklearn.ensemble import RandomForestClassifier

모델을 선언하여 객체화시킵니다.

In [51]:
rf_clf = RandomForestClassifier(
    n_estimators=200,     # 트리 개수
    max_depth=6,          # 너무 깊어지는 것 방지
    min_samples_split=10,
    min_samples_leaf=5,
    random_state=0,
    n_jobs=-1
)

모델을 학습 데이터에 맞춰 학습시킵니다.

In [53]:
rf_clf.fit(X_train, y_train)
rf_pred = rf_clf.predict(X_test)
accuracy_score(y_test, rf_pred)

0.67



<br/>
<br/>

# 6-3. 예측 성능 확인해보기 - RandomForestClassifier

- 학습된 모델로 테스트 데이터에 대한 예측을 수행합니다.

- 학습시킨 모델의 성능을 알아봅니다
- 각 평가지표로 모델의 성능을 수치화하여 확인합니다.
- 필요한 라이브러리를 import 하고 성능을 확인해보세요 (정확도, 정밀도, 재현율, f1, confusion_matrix)

In [58]:
print("📌 Random Forest 결과")
get_clf_eval(y_test, rf_pred)

📌 Random Forest 결과
Confusion Matrix
[[406  50]
 [181  63]]
------------------------------
Accuracy : 0.6700
Precision: 0.5575
Recall   : 0.2582
F1-score : 0.3529



<br/>

---

<br/>
<br/>

# 5-4. 모델링 - XGBoost

* 본격적으로 모델을 선언하고 학습시킵니다.



필요한 라이브러리를 불러옵니다.

In [60]:
from xgboost import XGBClassifier

모델을 선언하여 객체화시킵니다.

In [61]:
xgb_clf = XGBClassifier(
    n_estimators=300,
    learning_rate=0.05,
    max_depth=4,
    subsample=0.8,
    colsample_bytree=0.8,
    reg_lambda=1.0,
    random_state=0,
    eval_metric='logloss'   # 경고 방지용
)


모델을 학습 데이터에 맞춰 학습시킵니다.

In [63]:
xgb_clf.fit(X_train, y_train)
xgb_pred = xgb_clf.predict(X_test)
accuracy_score(y_test, xgb_pred)

0.6785714285714286



<br/>
<br/>

# 6-4. 예측 성능 확인해보기 - XGBoost

- 학습된 모델로 테스트 데이터에 대한 예측을 수행합니다.

- 학습시킨 모델의 성능을 알아봅니다
- 각 평가지표로 모델의 성능을 수치화하여 확인합니다.
- 필요한 라이브러리를 import 하고 성능을 확인해보세요 (정확도, 정밀도, 재현율, f1, confusion_matrix)

In [64]:
print("📌 XGBoost 결과")
get_clf_eval(y_test, xgb_pred)

📌 XGBoost 결과
Confusion Matrix
[[373  83]
 [142 102]]
------------------------------
Accuracy : 0.6786
Precision: 0.5514
Recall   : 0.4180
F1-score : 0.4755


<br/>

---


<br/>

## 7.  위 4가지 모델의 학습 & 예측 & 평가 결과를 확인하고 최고 성능을 내는 모델을 찾아봅시다!

- 어떤 모델이 가장 성능이 좋은가요 ?

In [66]:
from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score

def get_clf_eval(y_test, pred):
    cm = confusion_matrix(y_test, pred)
    acc = accuracy_score(y_test, pred)
    prec = precision_score(y_test, pred, zero_division=0)
    rec = recall_score(y_test, pred, zero_division=0)
    f1 = f1_score(y_test, pred, zero_division=0)

    print("Confusion Matrix")
    print(cm)
    print(f"Accuracy : {acc:.4f}")
    print(f"Precision: {prec:.4f}")
    print(f"Recall   : {rec:.4f}")
    print(f"F1-score : {f1:.4f}")

    return acc, prec, rec, f1

In [67]:
import pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from xgboost import XGBClassifier

results = []

# 1) Logistic Regression (표준화)
scaler = StandardScaler()
X_train_s = scaler.fit_transform(X_train)
X_test_s = scaler.transform(X_test)

lr = LogisticRegression(max_iter=2000, solver='liblinear', random_state=0)
lr.fit(X_train_s, y_train)
lr_pred = lr.predict(X_test_s)

print("\n📌 LogisticRegression")
acc, prec, rec, f1 = get_clf_eval(y_test, lr_pred)
results.append(["LogisticRegression", acc, prec, rec, f1])


# 2) Decision Tree (원본)
dt = DecisionTreeClassifier(
    max_depth=4,
    min_samples_split=10,
    min_samples_leaf=5,
    random_state=0
)
dt.fit(X_train, y_train)
dt_pred = dt.predict(X_test)

print("\n📌 DecisionTreeClassifier")
acc, prec, rec, f1 = get_clf_eval(y_test, dt_pred)
results.append(["DecisionTreeClassifier", acc, prec, rec, f1])


# 3) Random Forest (원본)
rf = RandomForestClassifier(
    n_estimators=200,
    max_depth=6,
    min_samples_split=10,
    min_samples_leaf=5,
    random_state=0,
    n_jobs=-1
)
rf.fit(X_train, y_train)
rf_pred = rf.predict(X_test)

print("\n📌 RandomForestClassifier")
acc, prec, rec, f1 = get_clf_eval(y_test, rf_pred)
results.append(["RandomForestClassifier", acc, prec, rec, f1])


# 4) XGBoost (원본)
xgb = XGBClassifier(
    n_estimators=300,
    learning_rate=0.05,
    max_depth=4,
    subsample=0.8,
    colsample_bytree=0.8,
    reg_lambda=1.0,
    random_state=0,
    eval_metric='logloss'
)
xgb.fit(X_train, y_train)
xgb_pred = xgb.predict(X_test)

print("\n📌 XGBClassifier")
acc, prec, rec, f1 = get_clf_eval(y_test, xgb_pred)
results.append(["XGBClassifier", acc, prec, rec, f1])


# 결과 표로 정리
result_df = pd.DataFrame(results, columns=["Model", "Accuracy", "Precision", "Recall", "F1"])
display(result_df.sort_values("F1", ascending=False))

best_model = result_df.sort_values("F1", ascending=False).iloc[0]
print(f"\n🏆 최고 성능 모델(F1 기준): {best_model['Model']}  |  F1 = {best_model['F1']:.4f}")


📌 LogisticRegression
Confusion Matrix
[[413  43]
 [177  67]]
Accuracy : 0.6857
Precision: 0.6091
Recall   : 0.2746
F1-score : 0.3785

📌 DecisionTreeClassifier
Confusion Matrix
[[358  98]
 [142 102]]
Accuracy : 0.6571
Precision: 0.5100
Recall   : 0.4180
F1-score : 0.4595

📌 RandomForestClassifier
Confusion Matrix
[[406  50]
 [181  63]]
Accuracy : 0.6700
Precision: 0.5575
Recall   : 0.2582
F1-score : 0.3529

📌 XGBClassifier
Confusion Matrix
[[373  83]
 [142 102]]
Accuracy : 0.6786
Precision: 0.5514
Recall   : 0.4180
F1-score : 0.4755


,Model,Accuracy,Precision,Recall,F1
3,XGBClassifier,0.678571,0.551351,0.418033,0.475524
1,DecisionTreeClassifier,0.657143,0.510000,0.418033,0.459459
0,LogisticRegression,0.685714,0.609091,0.274590,0.378531
2,RandomForestClassifier,0.670000,0.557522,0.258197,0.352941



🏆 최고 성능 모델(F1 기준): XGBClassifier  |  F1 = 0.4755


네 가지 분류 모델을 비교한 결과, 전반적으로 Accuracy는 0.65~0.69 수준으로 유사하게 나타났으나,<br>
Recall 및 F1-score에서는 차이가 관찰되었다.<br>
특히 XGBoost 모델이 F1-score 기준 가장 우수한 성능을 보였으며,<br>
이는 앙상블 기반 모델이 복잡한 비선형 패턴을 상대적으로 잘 포착했기 때문으로 해석된다.